api/src/main/java/org/eclipse/microprofile/jwt/JsonWebToken.java

```diff
@@ -31,8 +31,11 @@
  */
 public interface JsonWebToken extends Principal {
     /**
-     * Returns the unique name of this principal. This either comes from the upn claim, or if that is missing, the
-     * preferred_username claim. Note that for guaranteed interoperability a upn claim should be used.
+     * Returns the unique name of this principal. The upn claim is checked first, the preferred_username claim is
+     * checked next, and finally, the sub claim is checked. Note that for guaranteed interoperability a upn claim should
```

Comment:
We should consider adding a test in the TCK for this. Unless there's already one in there that I missed while looking just now.